hpctestlib/microbenchmarks/mpi/osu.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2016-2022 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+__all__ = ['fetch_osu_benchmarks', 'build_osu_benchmarks',
+           'osu_benchmark_test_base', 'osu_bandwidth', 'osu_latency']
+
+
+class fetch_osu_benchmarks(rfm.RunOnlyRegressionTest):
+    #: The version of OSU benchmarks to use.
+    #:
+    #: :type: :class:`str`
+    #: :default: ``'5.6.2'``
+    version = variable(str, value='5.9')
+
+    descr = 'Fetch OSU benchmarks'
+    local = True
+    executable = 'curl'
+
+    @run_after('init')
+    def set_executable_opts(self):
+        osu_file_name = f'osu-micro-benchmarks-{self.version}.tar.gz'
+        self.executable_opts = [
+            f'http://mvapich.cse.ohio-state.edu/download/mvapich/{osu_file_name}',  # noqa: E501
+            '--output', f'{osu_file_name}'
+        ]
+
+    @sanity_function
+    def validate_download(self):
+        return sn.assert_eq(self.job.exitcode, 0)
+
+
+class build_osu_benchmarks(rfm.CompileOnlyRegressionTest):
+    #: Option to build with support for accelerators
+    #:
+    #: :type: :class:`str`
+    #: :default: ``'cpu'``
+    build_type = parameter(['cpu', 'cuda', 'rocm', 'openacc'])
+
+    descr = 'Build OSU benchmarks'
+    build_system = 'Autotools'
+    build_prefix = variable(str)
+    osu_benchmarks = fixture(fetch_osu_benchmarks, scope='session')
+
+    @run_before('compile')
+    def prepare_build(self):
+        tarball = f'osu-micro-benchmarks-{self.osu_benchmarks.version}.tar.gz'
+        self.build_prefix = tarball[:-7]  # remove .tar.gz extension
+        fullpath = os.path.join(self.osu_benchmarks.stagedir, tarball)
+        self.prebuild_cmds += [
+            f'cp {fullpath} {self.stagedir}',
+            f'tar xzf {tarball}',
+            f'cd {self.build_prefix}'
+        ]
+        self.build_system.config_opts = [f'--enable-{self.build_type}']
+        self.build_system.make_opts = ['-C', 'mpi']
+        self.build_system.max_concurrency = 8
+
+    @sanity_function
+    def validate_build(self):
+        return True
```

Comment:
Perhaps a comment here as of why we seem not to validate here would help the reader understand better.